index.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+var analyze = require('./lib/analyze-raptor');
+var submitRegressions = require('./lib/submit-regressions');
+var Promise = require('promise');
+var fs = require('fs');
+
+var metric = 'visuallyLoaded';
+
+var readFile = function(dataFile) {
+  return new Promise(function(resolve, reject) {
+    fs.readFile(dataFile, { encoding: 'utf8' }, function(err, content) {
+      var data = content.split('\n');
+      data = JSON.parse(data);
+      if (err) {
+        reject(err);
+      } else {
+      	resolve(data);
+      }
+    });
+  });
+};
+
+var parseAppData = function(dataObj, appContext, appEntrypoint) {
+  return new Promise(function(resolve, reject) {
+    var perfData = [];
+    var series = dataObj.results[0].series;
+
+    for (entry in series) {
+      // only want measures
+      if (series[entry].name !== 'measure') {
+        continue;
+      }
+      // only want specific app
+      if (series[entry].tags.context !== appContext) {
+        continue;
+      }
+      if (appEntrypoint !== 'none' && series[entry].tags.enrypoint !== appEntrypoint) {
+        continue;
+      }
+      // only interested in specified metric ie. visuallyLoaded
+      if (series[entry].tags.metric !== metric) {
+        continue;
+      }
+
+      var revisionId = series[entry].tags.revisionId;
+      var values = series[entry].values;
+
+      // each measure may have multiple values [time, epoch, value]
+      // we want time, value, revisionId for each
+      for (nextValue in values) {
+        perfData.push({
+          time: values[nextValue][0],
+          value: values[nextValue][2],
+          revisionId: revisionId
+        });
+      }
+    }
+
+    resolve(perfData);
+  });
+};
+
+var raptorRegression = function(dataFile, appContext, appEntrypoint) {
+  require(dataFile);
+  console.log('Processing %s', dataFile);
+  console.log('App (context: %s, entrypoint: %s)', appContext, appEntrypoint);
+
+  readFile(dataFile)
+  .then(function(dataObj) {
+    return parseAppData(dataObj, appContext, appEntrypoint);
+  })
+  .then(function(perfData) {
+    if (perfData.length !== 0) {
```

Comment:
Actually, don't `catch` them here, bubble them all the way to the top script, then exit there.